ex0.py

```diff
@@ -0,0 +1,41 @@
+import functools
+
+
+class InvalidArgumentType(TypeError):
+    """Raise for invalid argument that passes to a function."""
+    __module__ = 'TypeError'
+
+
+def type_check(correct_type):
```

Comment:
Use type hints and return types -docum
https://www.geeksforgeeks.org/duck-typing-in-python/
https://stackoverflow.com/questions/125367/dynamic-type-languages-versus-static-type-languages
https://docs.python.org/3/library/typing.html
https://realpython.com/lessons/pros-and-cons-type-hints/